<launch>

  #### Nodelet manager ######################################################

  <node pkg="nodelet" type="nodelet" name="imu_manager3"
    args="manager" output="screen" >
 
    <remap from="imu/data_raw" to="imu3/data_raw" />
    <remap from="imu/data" to="imu3/data" />

  </node>

  #### IMU Driver ###########################################################

  <node pkg="nodelet" type="nodelet" name="PhidgetsImuNodelet3" 
    args="load phidgets_imu/PhidgetsImuNodelet imu_manager3" 
    output="screen">

    <remap from="imu/data_raw" to="imu3/data_raw" />
    <remap from="imu/data" to="imu3/data" />

    # supported data rates: 4 8 16 24 32 40 ... 1000 (in ms)
    <param name="period" value="4"/>
    
    <param name="frame_id" value="imu3"/>
    <param name="serial_number" value="297652"/>    
    
# compass correction params (see http://www.phidgets.com/docs/1044_User_Guide)
    <!-- <param name="cc_mag_field" value="0.52859"/>
         <param name="cc_offset0" value="0.03921"/>
         <param name="cc_offset1" value="0.19441"/>
         <param name="cc_offset2" value="-0.03493"/>
         <param name="cc_gain0" value="1.81704"/>
         <param name="cc_gain1" value="1.81028"/>
         <param name="cc_gain2" value="2.04819"/>
         <param name="cc_t0" value="0.00142"/>
         <param name="cc_t1" value="-0.03591"/>
         <param name="cc_t2" value="0.00160 "/>
         <param name="cc_t3" value="-0.05038"/>
         <param name="cc_t4" value="-0.03942"/>
         <param name="cc_t5" value="-0.05673"/> -->
  </node>


  #### IMU Orientation Filter ###############################################

  <node pkg="nodelet" type="nodelet" name="ImuFilterNodelet3" 
    args="load imu_filter_madgwick/ImuFilterNodelet imu_manager3" 
    output="screen">

    <param name="fixed_frame" value="imu_base" />
    <remap from="imu/data_raw" to="imu3/data_raw" />
    <remap from="imu/data" to="imu3/data" />

    <param name="use_mag" value="false"/>

  </node>


</launch>

